app/services/open_library_service.rb

```diff
@@ -0,0 +1,44 @@
+class OpenLibraryService
+  include HTTParty
+
+  base_uri Rails.application.credentials.open_library_uri
+  format :json
+
+  attr_accessor :isbn, :base_uri, :format, :jscmd
+
+  def self.fetch(isbn)
+    OpenLibraryService.new(isbn).book_info
+  end
+
+  def book_info
+    response = request_isbn
+    return nil unless isbn.start_with?('ISBN:')
+
+    cleanup(response.parsed_response[isbn])
+  end
+
+  private
+
+  def request_isbn
+    OpenLibraryService.get(base_uri, query: { bibkeys: isbn,
+                                              format: format,
+                                              jscmd: jscmd })
+  end
+
+  def cleanup(res)
+    json = {}
+    json[:ISBN] = isbn
+    json[:title] = res['title']
+    json[:subtitle] = res['subtitle']
+    json[:number_of_pages] = res['number_of_pages']
+    json[:authors] = res['authors']
+    json
+  end
+
+  def initialize(isbn)
```

Comment:
This should go to the top of your class. This is very good how you improve this class. To help you I will give the structure as tip:
```ruby
class OpenLibraryService
  include HTTParty
  base_uri Rails.application.secrets.API_BOOK_URL

  FORMAT = 'json'.freeze
  JSCMD = 'data'.freeze

  def initialize(isbn)
    @isbn = isbn
    @options = {
      query: {
        bibkeys: @isbn,
        format: FORMAT,
        jscmd: JSCMD
      }
    }
  end

  def book_info
    response = retrieve_from_api
    return {} if response.nil?

    format_response response
  end

  private

  def retrieve_from_api
    # Here execute the call with httparty
  end

  def format_response(response)
    # Here build you response
  end
end

```

Reply:
To call this in you controller do something like this:
```ruby
OpenLibraryService.new(isbn). book_info
```